linode/ansible_linode
# new: Support IP module to allocate a new IP

## 📝 Description

Support a new linode.cloud.ip module to allocate a new IP to a Linode instance.

## ✔️ How to Test

Integration test:
```
make TEST_ARGS="ip_basic" test
```

Manual test:

1. In a sandbox environment, i.e. dx-dvenv, run the following script to create a Linode instance and allocate a new IP to it.
```yaml
    - name: Create a Linode Instance
      linode.cloud.instance:
        label: 'ansible-instance-test'
        region: us-southeast
        type: g6-standard-1
        image: linode/alpine3.19
        state: present
      register: create_instance

    - name: Allocate a new IP to the Linode
      linode.cloud.ip:
        linode_id: '{{ create_instance.instance.id }}'
        public: true
        type: ipv4
        state: present
      register: allocate_ip
```

2. Observe the ip is allocated successfully.
3. Clean up the resources created.
